pkg/provision/hetzner.go

```diff
@@ -0,0 +1,158 @@
+package provision
+
+import (
+	"context"
+	"fmt"
+	"github.com/hetznercloud/hcloud-go/hcloud"
+	"strconv"
+)
+
+var Status = "running"
+
+type HetznerProvisioner struct {
+	client *hcloud.Client
+}
+
+// Creates a new Hetzner provisioner using an auth token.
+func NewHetznerProvisioner(authToken string) (*HetznerProvisioner, error) {
+	client := hcloud.NewClient(hcloud.WithToken(authToken))
+	return &HetznerProvisioner{
+		client: client,
+	}, nil
+}
+
+// Get status of a specific server by id.
+func (p *HetznerProvisioner) Status(id string) (*ProvisionedHost, error) {
+	intId, err := strconv.Atoi(id)
+	if err != nil {
+		return nil, err
+	}
+	server, _, err := p.client.Server.GetByID(context.Background(), intId)
+	if err != nil {
+		return nil, err
+	}
+	if server == nil {
+		return nil, fmt.Errorf("failed to find server with id %s", id)
+	}
+
+	status := ""
+	ip := ""
+
+	if server.Status == hcloud.ServerStatusRunning {
+		status = ActiveStatus
+		ip = server.PublicNet.IPv4.IP.String()
+	} else {
+		status = string(server.Status)
+	}
+
+	return &ProvisionedHost{
+		IP:     ip,
+		ID:     id,
+		Status: status,
+	}, nil
+}
+
+// Provision a new server on Hetzner cloud to use as an inlet node.
+func (p *HetznerProvisioner) Provision(host BasicHost) (*ProvisionedHost, error) {
+	if len(host.Plan) <= 0 {
+		host.Plan = "cx11"
+	}
+	if len(host.Region) <= 0 {
+		host.Region = "hel1"
+	}
```

Comment:
Setting Plan and Region to default here is not necessary. We should just fail it if for any reason the plan and region were not passed correctly from caller.